createmsi.py

```diff
@@ -261,6 +267,21 @@ def create_registry_entries(self, comp, reg):
             'KeyPath': reg['key_path'],
           })
 
+    def create_custom_actions(self, product, install_execute_sequence, action):
+        ET.SubElement(product, 'CustomAction', {
+            'Id': action['id'],
+            'Property': action['property'],
+            'ExeCommand': action['exe_command'],
+            'Execute': action['execute'],
+            'Return': action['return'],
+            'Impersonate': action['impersonate'],
+        })
+        key = 'After' if action.has_key('after') else 'Before'
```

Comment:
This fails to run. Dicts do not have a `has_key` method. The correct form is `if 'after' in action`.

Reply:
I was using python 2.7. I have changed to use the new way.